packages/builder/src/builder.js

```diff
@@ -297,100 +220,118 @@ export default class Builder {
     // Suggest dependency fixes (TODO: automate me)
     if (dpendencyFixes.length) {
       consola.error(
-        `Please install missing dependencies:\n`,
+        'Please install missing dependencies:\n',
         '\n',
-        `Using yarn:\n`,
+        'Using yarn:\n',
         `yarn add ${dpendencyFixes.join(' ')}\n`,
         '\n',
-        `Using npm:\n`,
+        'Using npm:\n',
         `npm i ${dpendencyFixes.join(' ')}\n`
       )
       throw new Error('Missing Template Dependencies')
     }
   }
 
   async generateRoutesAndFiles() {
-    consola.debug(`Generating nuxt files`)
+    consola.debug('Generating nuxt files')
 
     // Plugins
     this.plugins = Array.from(this.normalizePlugins())
 
-    // -- Templates --
-    let templatesFiles = Array.from(this.template.files)
-
-    const templateVars = {
-      options: this.options,
-      extensions: this.options.extensions
-        .map(ext => ext.replace(/^\./, ''))
-        .join('|'),
-      messages: this.options.messages,
-      splitChunks: this.options.build.splitChunks,
-      uniqBy,
-      isDev: this.options.dev,
-      isTest: this.options.test,
-      debug: this.options.debug,
-      vue: { config: this.options.vue.config },
-      mode: this.options.mode,
-      router: this.options.router,
-      env: this.options.env,
-      head: this.options.head,
-      store: this.options.store,
-      globalName: this.options.globalName,
-      globals: this.globals,
-      css: this.options.css,
-      plugins: this.plugins,
-      appPath: './App.js',
-      layouts: Object.assign({}, this.options.layouts),
-      loading:
-        typeof this.options.loading === 'string'
-          ? this.relativeToBuild(this.options.srcDir, this.options.loading)
-          : this.options.loading,
-      transition: this.options.transition,
-      layoutTransition: this.options.layoutTransition,
-      dir: this.options.dir,
-      components: {
-        ErrorPage: this.options.ErrorPage
-          ? this.relativeToBuild(this.options.ErrorPage)
-          : null
-      }
-    }
+    const templateContext = new TemplateContext(this, this.options)
+
+    await this.resolveLayouts(templateContext)
+    await this.resolveRoutes(templateContext)
+    await this.resolveStore(templateContext)
+    await this.resolveMiddleware(templateContext)
+    await this.resolveCustomTemplates(templateContext)
+    await this.resolveLoadingIndicator(templateContext)
```

Comment:
Would `Promise.all` be useful here?

Reply:
Yes it's a good idea, but the order of the resolve method may have dependencies between each other.

Some methods may change the `options` or member variables which are used in next methods, we may need to  go through all the methods and refactor them

Reply:
I'll do it now.